parsers/CA_YT.py

```diff
@@ -1,128 +1,119 @@
-
-    # generation is specified like "37.69 MW - hydro"
-    hydro_div = find_div_by_class(soup, "load_hydro")
-    hydro_text = hydro_div.div.text
-    hydro_generation = parse_mw(hydro_text)
-
-    hydro_cap_div = find_div_by_class(soup, "avail_hydro")
-    if hydro_cap_div:
-        hydro_cap_text = hydro_cap_div.div.text
-        hydro_capacity = parse_mw(hydro_cap_text)
-    else:
-        # hydro capacity is not provided when thermal is used
-        hydro_capacity = None
-
-    thermal_div = find_div_by_class(soup, "load_thermal")
-    if thermal_div.div:
-        thermal_text = thermal_div.div.text
-        thermal_generation = parse_mw(thermal_text)
-    else:
-        # thermal is not always used and when it's not used, it's not specified in HTML
-        thermal_generation = 0
-
-    data = {
-        "datetime": datetime_datetime,
-        "zoneKey": zone_key,
-        "production": {
-            "unknown": thermal_generation,
-            "hydro": hydro_generation,
-            # specify some sources that aren't present in Yukon as zero,
-            # this allows the analyzer to better estimate CO2eq
-            "coal": 0,
-            "nuclear": 0,
-            "geothermal": 0,
-        },
-        "storage": {},
-        "source": "www.yukonenergy.ca",
-    }
-
-    if hydro_capacity:
-        data.update({"capacity": {"hydro": hydro_capacity}})
-
-    return data
+    if zone_key != ZONE_KEY:
+        raise ParserException("CA_YT.py", "Cannot parse zone '{zone_key}'", zone_key)
+    if target_datetime:
+        raise ParserException("CA_YT.py", "Unable to fetch historical data", zone_key)
+
+    session = session or Session()
+    soup = BeautifulSoup(session.get(URL).text, "html.parser")
+
+    # Extract the relevant HTML data.
+    # The date is formatted as, e.g., "Thursday, June 22, 2017".
+    date = soup.find("div", class_="current_date").text
+    # The time is formatted as, e.g., "11:55 pm" or "2:25 am".
+    time = soup.find("div", class_="current_time").text
+    # Note: hydro capacity is not provided when thermal is in use.
+    hydro_capacity = soup.find("div", class_="avail_hydro")
+    thermal = soup.find("div", class_="load_thermal").div
+
+    production_mix = ProductionMix()
+    production_mix.add_value("coal", 0)
+    production_mix.add_value("geothermal", 0)
+    production_mix.add_value(
+        "hydro", _parse_mw(soup.find("div", class_="load_hydro").div.text)
+    )
+    production_mix.add_value("nuclear", 0)
+    production_mix.add_value("unknown", _parse_mw(thermal.text) if thermal else 0)
+
+    production_breakdowns = ProductionBreakdownList(logger=logger)
+    production_breakdowns.append(
+        datetime=datetime.strptime(f"{date} {time}", "%A, %B %d, %Y %I:%M %p").replace(
+            tzinfo=TIMEZONE
+        ),
+        production=production_mix,
+        source=SOURCE,
+        storage=StorageMix(
+            hydro=_parse_mw(hydro_capacity.div.text) if hydro_capacity else None
+        ),
```

Comment:
Why is this added to storage mix?

This was originally added to the capacity object (this is not supported by the parser classes though)

Reply:
Good question... I think I thought they were the same for some reason. I guess this should be removed! :grimacing:

Reply:
I see you've fixed this in another PR, thanks!